### PR TITLE
test_2238777_LiveMigrationOfVMsWithDataOnRootAndEphemeralDisk

### PR DESCRIPTION
**Pull request by tkuterina**
New TC test_2238777_LiveMigrationOfVMsWithDataOnRootAndEphemeralDisk    was added
Common function "create_instance" was updated with new parameter "inst_list" (default value is None) 
Common functions "is_key_exists" and "delete_keys" were added

### CODE REVIEW

File: mos_tests/functions/common.py [hunk 1]
```diff
@@ -543,3 +548,27 @@ def delete_volume_snapshot(cinder_client, snapshot):
...
+
+
+# Keys
+def is_key_exists(nova_client, key_name):
+    """ Check the presence of keys in the system
+        :param nova_client: Nova API client connection point
+        :param key_name: name of the keypair
+        :return True or False
+    """
+    return key_name in [k.name for k in nova_client.keypairs.list()]
```
Comment (gdyuldin) [inline mos_tests/functions/common.py]: Nova client has method 'find'

Comment (tkuterina) [inline mos_tests/functions/common.py]: Method 'find' returns keypair or raise NotFound exception, we need only true if keypair exists or false if not.

Comment (gdyuldin) [inline mos_tests/functions/common.py]: Ok. But you can use method `findall`. It returns empty list if there are no keys with this name.

File: mos_tests/functions/common.py [hunk 2]
```diff
@@ -301,10 +301,12 @@ def delete_instance(nova_client, uid):
             sleep(1)
 
 
-def create_instance(nova_client, inst_name, flavor_id, net_id, security_groups,
-                    image_id='', block_device_mapping=None, timeout=5):
+def create_instance(nova_client, inst_list, inst_name, flavor_id, net_id,
```
Comment (MikeLaptev) [inline mos_tests/functions/common.py]: Did you check all tests that uses this function?
And why we need such parameter in common function?
Why test cannot manage this list by test itself?

Comment (tkuterina) [inline mos_tests/functions/common.py]: Misha, several days ago you said me, that my test don't delete instance if during instance creation status isn't ACTIVE and exception raises. For this case inst_list was added in  parameter's list for create_instance common function. I update and check tests in nova_tests.py file. Test can manage this list itself, but try-exept-finaly construction should be added in this common function for return id, or in all tests which used this function

File: mos_tests/functions/common.py (hunk 1 above)
Comment (tkuterina) [inline mos_tests/functions/common.py]: This function is similar to other such functions in this file (is_volume_exists, is_instance_exists, etc)

File: mos_tests/functions/common.py (hunk 2 above)
Comment (MikeLaptev) [inline mos_tests/functions/common.py]: Yes, you are right, about the moment when I mentioned that, but what about rest of tests that depends on common.py? 
Did you check all of them?

In any case, let's wait for Timur's oppinion about this parameter, and mostly about right workflow for that case.

**Comment by MikeLaptev** [comment]
In common case, would you be so kind to add brief description of your changes in pull request, please?
Just to avoid following string on top of it 'No description provided.'

File: mos_tests/functions/common.py (hunk 1 above)
Comment (gdyuldin) [inline mos_tests/functions/common.py]: Performance of retrieving all data via http rest API and filtering it later will be worst than filtering data on the nova side. And IMHO 
`return len(nova_client.keypairs.findall(name=key_name))) > 0`
looks clearer than
`return key_name in [k.name for k in nova_client.keypairs.list()]`

File: mos_tests/nova/nova_tests.py [hunk 3]
```diff
@@ -466,3 +484,106 @@ def test_2238776_NetworkConnectivityToVMDuringLiveMigration(self):
...
+            inst = self.nova.servers.get(inst.id)
+        self.assertEqual(inst.status, 'ACTIVE')
+        ping.wait()
+        output = ping.stdout.read().split('\n')[-3].split()
+        packets = {'transmitted': int(output[0]), 'received': int(output[3])}
+        loss = packets['transmitted'] - packets['received']
+        if loss > 5:
+            raise AssertionError(
+                "Packets loss exceeds the limit, {} packets were lost".format(
+                     loss))
```
Comment (TimurNurlygayanov) [inline mos_tests/nova/nova_tests.py]: msg = "Packets loss exceeds the limit, {} packets were lost"
raise AssertionError(msg.format(loss))

File: mos_tests/nova/nova_tests.py [hunk 4]
```diff
@@ -466,3 +484,106 @@ def test_2238776_NetworkConnectivityToVMDuringLiveMigration(self):
...
+        out = []
+        with SSHClient(host=floating_ip.ip, username="cirros", password=None,
+                       private_keys=[private_key]) as vm_r:
+            out.append(vm_r.execute("sudo -i cat /timestamp.txt"))
+            out.append(vm_r.execute("sudo -i cat /mnt/timestamp.txt"))
+
+        for i in out:
+            if i['stderr'] != []:
```
Comment (TimurNurlygayanov) [inline mos_tests/nova/nova_tests.py]: if i['stderr'] != []:    >  i.get('stderr'):

File: mos_tests/nova/nova_tests.py [hunk 5]
```diff
@@ -466,3 +484,106 @@ def test_2238776_NetworkConnectivityToVMDuringLiveMigration(self):
...
+        out = []
+        with SSHClient(host=floating_ip.ip, username="cirros", password=None,
+                       private_keys=[private_key]) as vm_r:
+            out.append(vm_r.execute("sudo sh -c 'date > /timestamp.txt'"))
+            out.append(vm_r.execute("sudo sh -c 'date > /mnt/timestamp.txt'"))
+            out.append(vm_r.execute("sudo -i cat /timestamp.txt"))
+            out.append(vm_r.execute("sudo -i cat /mnt/timestamp.txt"))
+
+        for i in out:
+            if i['stderr'] != []:
```
Comment (TimurNurlygayanov) [inline mos_tests/nova/nova_tests.py]: if i['stderr'] != []: > i.get('stderr'):
